### PR TITLE
Fixes #116

### LINKED ISSUE #116: Adopt the Integration Cookbook
Convert, revise, and eventually publish the Integration Cookbook. See https://github.com/openshiftio/openshift.io/issues/1324

Currently available from https://docs.google.com/document/d/1k5hbknuebBZVVMkWpKZKAP6LoHWXD2bbvrn-4P4wmjg/

### PR DESCRIPTION
**Pull request by Preeticp**
The Integration cookbook has been converted from google doc to asciidoc.
I have taken a pass at modularizing the content and creating assemblies.
Please note, content has not been changed, as yet, from the original, only structure, format changes post adoc conversion, and modularization has been done in this first commit.

Done:
- Review and rewrite content. 
- Add content.
- Update screenshots if required.

### CODE REVIEW

**Comment by Preeticp** [comment]
To be merged only after this sprint.

**Comment by fabric8cd** [comment]
@Preeticp snapshot fabric8/fabric8-online-docs:SNAPSHOT-PR-149-2 is available for testing at http://fabric8-online-docs-app-pr-149-fabric8-online-docs.badger.fabric8.io

**Comment by fabric8cd** [comment]
@Preeticp snapshot fabric8/fabric8-online-docs:SNAPSHOT-PR-149-4 is available for testing at http://fabric8-online-docs-app-pr-149-fabric8-online-docs.badger.fabric8.io

**Comment by Preeticp** [comment]
@rkratky we can go ahead and publish this. A note has been added to contact folks on the mattermost channel/email OpenShift.io for the resource URL as requested by the Analytics team. They do not want to change the name for the guide.
The validate and build scripts won't be relevant for this, correct me if I am wrong.

File: docs/topics/case-studies.adoc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[id="case_studies"]
+= Case Studies
+
+The following section lists a couple of case studies for the Fabric8-Analytics platform.
```
Comment (rkratky) [inline docs/topics/case-studies.adoc]: a couple --> two

File: docs/topics/modules/about-analytics-integration-cookbook.adoc [hunk 2]
```diff
@@ -0,0 +1,18 @@
+[id="about_analytics_integration_cookbook"]
+= About Fabric8-Analytics integration cookbook
+
+Fabric8-Analytics is an open source analytics platform aimed primarily at assisting developers in getting insights and recommendations on the stacks used for developing applications. This platform can be used to get an overview and recommendations for single application components as well.
+
+This cookbook is written for developers who are interested in integrating https://github.com/fabric8-analytics/[Fabric8-Analytics] _stack-analyses_ or _component-analyses_ services within their applications.
+
+The integration cookbook runs you through a couple of case-studies, which describe its usage and describes the other API resources you can call to use Fabric8-Analytics.
```
Comment (rkratky) [inline docs/topics/modules/about-analytics-integration-cookbook.adoc]: a couple --> two

Comment (rkratky) [inline docs/topics/modules/about-analytics-integration-cookbook.adoc]: runs you --> guides you, introduces you to, ...

Generally, do not use phrasal verbs in docs.

File: docs/topics/modules/about-analytics-integration-cookbook.adoc [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+== Supported Ecosystems
+
+Fabric8-Analytics _stack-analyses_ analyzes application stacks of the following ecosystems currently:
```
Comment (rkratky) [inline docs/topics/modules/about-analytics-integration-cookbook.adoc]: Word order --> F8-Ana _currently_ analyzes...

File: docs/topics/modules/api-get-component-analyses-response.adoc [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+| Requires Authentication | Yes
+|===
+
+== Parameters
+
+None
+
+== Example Request with cURL
+
+[source,typescript]
```
Comment (rkratky) [inline docs/topics/modules/api-get-component-analyses-response.adoc]: This is as command, not typescript source code.

File: docs/topics/modules/api-get-component-analyses-response.adoc [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+----
+
+*Failure*
+
+[source,typescript]
+----
+401:{  "error": "Authentication failed - could not decode JWT token"}404:{  "error": "No data found for maven Package io.vertx:vertx-core/3.4.1"}
+[source,typescript]
```
Comment (rkratky) [inline docs/topics/modules/api-get-component-analyses-response.adoc]: I'm guessing this line shouldn't be here.

File: docs/topics/modules/api-get-stack-analyses-response.adoc [hunk 6]
```diff
@@ -0,0 +1,53 @@
+[id="api_get_stack_analyses_response"]
+= GET /stack-analyses/{request ID}
```
Comment (rkratky) [inline docs/topics/modules/api-get-stack-analyses-response.adoc]: Escape.

File: docs/topics/modules/api-get-component-analyses-response.adoc [hunk 7]
```diff
@@ -0,0 +1,46 @@
+[id="api_get_component_analyses_response"]
+= GET /component-analyses/{ecosystem}/{component name}/{component version}
+
+== Purpose
+Fetches various data points for a selected component.
+
+IMPORTANT: You require an {osio} authentication token to invoke this method.
```
Comment (rkratky) [inline docs/topics/modules/api-get-component-analyses-response.adoc]: This doesn't seem like an 'important' admonition...

* note -- a reminder (such as this case)
* important -- potential breakage if not heeded
* warning -- danger of loss of data

File: docs/topics/modules/api-get-stack-analyses-response.adoc [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+== Purpose
+
+Fetches stack analysis response by using the request identifier returned by /stack-analyses POST.
+
+IMPORTANT: You require  an {osio} authentication token to invoke this method.
```
Comment (rkratky) [inline docs/topics/modules/api-get-stack-analyses-response.adoc]: See above.

File: docs/topics/modules/api-get-stack-analyses-response.adoc [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+== Resource URL
+
+For information on the resource URL reach out to us on the {osio} support chat channel at link:https://chat.openshift.io[https://chat.openshift.io] or email us at link:mailto:openshiftio@redhat.com[openshiftio@redhat.com].
```
Comment (rkratky) [inline docs/topics/modules/api-get-stack-analyses-response.adoc]: --> ... resource URL, reach...

File: docs/topics/modules/api-get-stack-analyses-response.adoc [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+| Requires Authentication | Yes
+|===
+
+== Parameters
+
+None
+
+== Example Request with cURL
+
+[source,typescript]
```
Comment (rkratky) [inline docs/topics/modules/api-get-stack-analyses-response.adoc]: Not source code.

File: docs/topics/modules/api-get-stack-analyses-response.adoc [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
..."EPL 1.0",                    "Apache 2.0",                    "MIT License"                ],                "ecosystem": "maven",                "license_analysis": {                    "conflict_packages": [],                    "f8a_stack_licenses": [],                    "outlier_packages": [],                    "status": "Unknown",                    "unknown_licenses": {                        "component_conflict": [],                        "really_unknown": [                            {                                "license": "EPL 1.0",                                "package": "io.vertx:vertx-core"                            },                            {                                "license": "EPL 1.0",                                "package": "io.vertx:vertx-web"                            }                        ]                    }                },                "recommendation_ready": true,                "stack_license_conflict": true,                "total_licenses": 3,                "unknown_dependencies": [],                "unknown_dependencies_count": 0            }        }    ],    "started_at": "2017-10-20T05:09:35.182243",    "version": "v1"}
```
Comment (rkratky) [inline docs/topics/modules/api-get-stack-analyses-response.adoc]: See above.

File: docs/topics/modules/api-post-stack-analyses-request.adoc [hunk 12]
```diff
@@ -0,0 +1,53 @@
+[id="api_post_stack_analyses_request"]
+= POST /stack-analyses
+
+== Purpose
+
+Submits an application stack analysis request
+
+IMPORTANT: You require an {osio} authentication token to invoke this method.
```
Comment (rkratky) [inline docs/topics/modules/api-post-stack-analyses-request.adoc]: See above.

File: docs/topics/modules/api-post-stack-analyses-request.adoc [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+
+
+
+== Resource URL
+
+For information on the resource URL reach out to us on the {osio} support chat channel at link:https://chat.openshift.io[https://chat.openshift.io] or email us at link:mailto:openshiftio@redhat.com[openshiftio@redhat.com].
```
Comment (rkratky) [inline docs/topics/modules/api-post-stack-analyses-request.adoc]: See above.

File: docs/topics/modules/api-post-stack-analyses-request.adoc [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+| filePath[]                                                                                                                        | required                                                                                                                          | Corresponding manifest file’s directory information. Care has to be taken to map the manifest filenames and file paths correctly. | /home/JohnDoe
+| origin                                                                                                                            | optional                                                                                                                          | Originator of the request. Can be the name of an application integrating to the endpoint. Maintained for historical reasons.      | "my-app"
+|===
+
+== Example Request with cURL
+
+[source,typescript]
```
Comment (rkratky) [inline docs/topics/modules/api-post-stack-analyses-request.adoc]: See above.

File: docs/topics/modules/case-study-1.adoc [hunk 15]
```diff
@@ -0,0 +1,35 @@
+[id="case_study_1"]
+= Case Study 1: Detecting security vulnerabilities associated with a component
+
+This scenario can be addressed by calling GET `/component-analyses` method and parsing the result for security vulnerabilities.
+Following is a typescript code snippet to invoke `/component-analyses` method:
```
Comment (rkratky) [inline docs/topics/modules/case-study-1.adoc]: --> The following is...

-> to invoke the `/component-analyses` method

File: docs/topics/modules/case-study-1.adoc [hunk 16]
```diff
@@ -0,0 +1,35 @@
+[id="case_study_1"]
+= Case Study 1: Detecting security vulnerabilities associated with a component
+
+This scenario can be addressed by calling GET `/component-analyses` method and parsing the result for security vulnerabilities.
```
Comment (rkratky) [inline docs/topics/modules/case-study-1.adoc]: --> by calling the GET...

File: docs/topics/modules/getting-osio-authentication-token.adoc [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+= Getting OpenShift.io authentication token
+
+You require an {osio} authentication token to invoke all Fabric8-Analytics API resources.
+
+Get the {osio} authentication token as follows:
+
+. Log into link:https://openshift.io[{osio}]
+. Open the user profile.
++
+image::profile.png[user profile]
```
Comment (rkratky) [inline docs/topics/modules/getting-osio-authentication-token.adoc]: We don't need to do the red circles outside of the GSG. And we'll be getting rid of them even there. In fact, we don't want them if at all possible.

File: docs/topics/modules/getting-osio-authentication-token.adoc [hunk 18]
```diff
@@ -0,0 +1,21 @@
...
++
+. Click on Update Profile
```
Comment (rkratky) [inline docs/topics/modules/getting-osio-authentication-token.adoc]: --> `btn:[Update Profile]`

File: docs/topics/modules/getting-osio-authentication-token.adoc [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
++
+image::update_profile.png[update profile]
++
+. Copy the Access Token from the Update Profile page.
```
Comment (rkratky) [inline docs/topics/modules/getting-osio-authentication-token.adoc]: --> `*Update Profile*`

File: docs/topics/modules/getting-osio-authentication-token.adoc (hunk 17 above)
Comment (rkratky) [reply docs/topics/modules/getting-osio-authentication-token.adoc]: Also, images should be named better -- to make it possible to recognize where they belong anmd what's on them from the name itself. `profile.png` is very vague.

@mishaone FYI, too.

File: docs/topics/modules/api-get-stack-analyses-response.adoc (hunk 11 above)
Comment (rkratky) [reply docs/topics/modules/api-get-stack-analyses-response.adoc]: Also, such large piece of source code should be included as a separate file to make the module file easier to view.

**Comment by rkratky** [comment]
Completed by #255.

**Pull request closed without merging**